docs/decisions/0004-iframe-credit-card-handling.rst

```diff
@@ -0,0 +1,47 @@
+4. iFrame Credit Card Handling
+------------------------------
+
+Status
+------
+
+Draft
+
+Context
+-------
+
+Credit card numbers and security codes are very sensitive classes of data. They are closely protected by the Payment Card Industry Data Security Standard (PCI DSS). For merchants to process credit cards they must be compliant with this standard. Unless transactions volumes are very high, the PCI DSS does not require an external certification, instead requiring merchants to complete a Self-Assessment Questionnaire (SAQ). The simplest version of this questionnaire is only applicable if the merchant completely outsources the handling, storage and processing of the sensitive credit card data to a PCI compliant vendor. This requires the vendor hosting all data entry elements. A common practice used by payment vendors is to offer an iframe integration where the merchant controls the checkout experience, however, the sensitive fields are injected into the flow via an iframes. Browser security controls prevent the host page from being able to access the sensitive data within the iframes. When the user is ready to checkout, the sensitive data is converted into an opaque token that is then given to the merchant to authorize the payment. This architecture has been broadly adopted by the industry and is used by major vendors like Stripe and Baintree to reduce the compliance burden for merchants integrating their payment widgets. This method dramatically reduces the risk of malicious open-source third-party dependencies harvesting credit card numbers, or accidental leakage of sensitive credit card data by developers making changes to the software since the code running on frontend-app-payment no longer has access to the sensitive data.
+
+Cybersource also provides an integration method called "Silent Order Post" (SOP) that allows merchants to put a <form> element on the basket page that directly posts to Cybersource allowing the merchant to avoid sending the sensitive data to back-end servers. This has been the method that frontend-app-payment has been using since it was initially created.
+
+Cybersource also provides an SDK called the "Flex Microform SDK" that allows merchants (like Open edX sites) to easily use the iframe integration method to embed the sensitive widgets directly into existing basket pages in a way that is transparent to the user.
+
+Decision
+--------
+
+frontend-app-payment will now use the Flex Microform SDK to handle credit card payments instead of SOP.
```

Comment:
We should also mention a) we aren't doing any more refactoring right now, and b) we didn't convert all uses of SOP, just the primary payment flow.